sa/sa.go

```diff
@@ -351,6 +351,11 @@ func (ssa *SQLStorageAuthority) NewOrderAndAuthzs(ctx context.Context, req *sapb
 			}
 		}
 
+		// Prevent a panic if the inner NewOrder object doesn't exist
+		if req.NewOrder == nil {
+			return nil, errIncompleteRequest
+		}
```

Comment:
I would do this check at the very top, between lines 313 and 314, so that we don't even start the transaction in this case.